### PR TITLE
chore(deps): replace didyoumean with leven

### PR DESCRIPTION
**Pull request by jamesgeorge007**
**What**:

Currently, the codebase depends on [didyoumean](https://www.npmjs.com/package/didyoumean) to suggest the closest match to an unknown command. This PR aims at making use of a [better alternative](https://www.npmjs.com/package/leven) for the purpose.

**Why**:

[didyoumean](https://www.npmjs.com/package/didyoumean) isn't maintained anymore; hence it is better to migrate to [leven](https://www.npmjs.com/package/leven).

**How**:

Replaced `didyoumean` with `leven`.

**Tasks**:

- [x] Code
- [x] Add a changeset (with link to its [Feature Discussion](https://community.frontity.org/c/33) if it exists)

**Unrelated Tasks**

- [ ] TSDocs
- [ ] TypeScript
- [ ] Unit tests
- [ ] End to end tests
- [ ] TypeScript tests
- [ ] Update starter themes
- [ ] Update other packages
- [ ] Open an issue for this feature in the [docs repo](https://github.com/frontity/docs/wiki/Code-Releases)
- [ ] Update community discussions

**Additional Comments**

N/A

### CODE REVIEW

**Review by DAreRodz — CHANGES_REQUESTED** [review]
Hi @jamesgeorge007, good to see you want to contribute to Frontity :blush: 

You are right, `didyoumean` has not being maintained for a long time, but the library you propose doesn't do exactly the same - it only returns the Levenshtein distance - , so we would have to write some code to get the suggested command.

I requested some changes. :point_down:

File: packages/frontity/src/cli/unknown.ts [hunk 1]
```diff
@@ -12,7 +12,9 @@ const unknown = (command: string, program: CommanderStatic) => {
   console.log(chalk.red(`Unknown command: ${chalk.bold(command)}`));
 
   const availableCommands: string[] = program.commands.map((c) => c._name);
-  const suggestion: string | string[] = didYouMean(command, availableCommands);
+  const suggestion: string = availableCommands.find(
+    (c) => leven(c, command) < 3
```
[the remark this answers is not in the document]
Comment (jamesgeorge007) [reply packages/frontity/src/cli/unknown.ts]: It should be better now.

File: packages/frontity/src/cli/unknown.ts [hunk 2]
```diff
@@ -12,7 +12,9 @@ const unknown = (command: string, program: CommanderStatic) => {
   console.log(chalk.red(`Unknown command: ${chalk.bold(command)}`));
 
   const availableCommands: string[] = program.commands.map((c) => c._name);
-  const suggestion: string | string[] = didYouMean(command, availableCommands);
+  const suggestion: string = availableCommands.find(
+    (c) => leven(c, command) < c.length * 0.4
```
Comment (DAreRodz) [inline packages/frontity/src/cli/unknown.ts]: OK, that's a better threshold, @jamesgeorge007 . :+1:

Now, I think we should return the one with the least distance under that threshold, something like this:

```ts
const [suggestion] = availableCommands
  // Compute the distance of each command and return it along with the name.
  .map((name) => ({ name, distance: leven(name, command) }))
  // Filter commands by threshold.
  .filter(({ name, distance }) => distance < name.length * 0.4)
  // Sort them so the one with the least distance would be the first one.
  .sort((c1, c2) => c1.distance - c2.distance)
  // Map them and return only the command name.
  .map((c) => c.name)
```

That way we would be destructuring the first element of the available commands, filtered by the threshold and sorted by distance.

BTW, you can go to any of the example projects in the monorepo - for example, `/examples/mars-theme-example/` -and test the `npx frontity` commands there. :slightly_smiling_face:

**Review by DAreRodz — APPROVED** [review]
I tested it and it works fine. Thanks for the PR, @jamesgeorge007. :blush: 

BTW, if you want to keep contributing, you can take a look at our issues labeled with [good first issue](https://github.com/frontity/frontity/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22).

**Pull request merged**